### PR TITLE
Document limited uri-template support

### PR DESCRIPTION
**Pull request by matthias-pichler-warrify**
Signed-off-by: Matthias Pichler <m.pichler@warrify.com>

**Please specify parts of this PR update:**

- [x] Specification
- [ ] Schema
- [ ] Examples
- [ ] Extensions
- [ ] Use Cases
- [ ] Community
- [ ] CTK
- [ ] Other

**Discussion or Issue link**:

As discussed in #910

**What this PR does**:

Documents the frequently used by undocumented string substitution feature

**Additional information:**

### CODE REVIEW

**Comment by cdavernas** [comment]
I'm not sure string substitution should be allowed everywhere, but instead should be restricted to URIs, else it would ask too much checks at runtime, don't you think?

**Comment by matthias-pichler-warrify** [comment]
> I'm not sure string substitution should be allowed everywhere, but instead should be restricted to URIs, else it would ask too much checks at runtime, don't you think?

actually I think it should be available everywhere where a runtime expression could be used (so pretty much everywhere 😅 ).
Furthermore most uris are currently defined as [`uri-template`](https://datatracker.ietf.org/doc/html/rfc6570) in the schema ... are wqe planning on supporting the full uri-template syntax or only the simple substitution?

**Comment by cdavernas** [comment]
> actually I think it should be available everywhere where a runtime expression could be used

That's why I'd personally refrain from doing so, as you can easily do that using runtime expressions, which are much easier to spot for a runtime. 

This would imho become extremely cumbersome to have to check for both `${}` blocks and `{}` blocks, which could in addition conflict. 

Furthermore, it would mean authors would not be able to use those characters without forcing runtimes to evaluate, even when not necessary, therefore needing some escaping feature.

Anyhow, thats just me, let's see what @JBBianchi, @fjtirado and @ricardozanini think about it 😉

**Comment by cdavernas** [comment]
> Furthermore most uris are currently defined as [uri-template](https://datatracker.ietf.org/doc/html/rfc6570) in the schema ... are wqe planning on supporting the full uri-template syntax or only the simple substitution?

IMHO only simple substitution

**Comment by matthias-pichler-warrify** [comment]
> That's why I'd personally refrain from doing so, as you can easily do that using runtime expressions, which are much easier to spot for a runtime.
>
> This would imho become extremely cumbersome to have to check for both ${} blocks and {} blocks, which could in addition conflict.
>
> Furthermore, it would mean authors would not be able to use those characters without forcing runtimes to evaluate, even when not necessary, therefore needing some escaping feature.

No strong opinion from my side. I am fine with only allowing this in URIs. In this case though I would rephrase the feature away from "string substitution" to "limited support for uri-template syntax". This would then allow us to expand to support additional uri-template features in the future (or now?).

**Review by cdavernas — CHANGES_REQUESTED** [review]
LGTM, aside from two little typos ;) Many thanks!

**Comment by fjtirado** [comment]
Although already changed, I agree with @cdavernas. String substitution should apply only to uri templates

**Review by fjtirado — APPROVED** [review]

**Review by cdavernas — APPROVED** [review]
LGTM! Thanks man!

**Review by ricardozanini — APPROVED** [review]
